jasmine/src/Game.js

```diff
@@ -0,0 +1,84 @@
+function Game() {
+  this.currentFrameIndex = 0;
+  this.frames = new Array(10).fill(null).map(function() {
+    return [];
+  });
+};
+
+Game.prototype.score = function () {
+  return this.frames.reduce(function(totalScore, frame) {
+    return totalScore + frame.reduce(function(acc, pins) { return acc + pins }, 0)
+  }, 0)
+}
+
+Game.prototype.currentFrame = function(){
+  return this.frames[this.currentFrameIndex]
+}
+
+Game.prototype.bowl = function(pins) {
+  if (this.currentFrameIndex === 9) {
+    this.processFinalFrame(pins);
+  } else {
+    this.processFrame(pins);
+  }
+}
+
+Game.prototype.processFrame = function(pins) {
+  if (this.isPinsNotAllowed(pins)) {
+    throw new Error ("Illegal");
+  }
+  this.currentFrame().push(pins);
+  this.processPreviousFrameBonus(pins);
+  this.processPreviousPreviousFrameBonus(pins);
+  if (this.isCurrentFrameComplete()) {
+    this.currentFrameIndex += 1;
+  }
+};
+
+Game.prototype.processFinalFrame = function(pins){
+  if (this.isFinalFrameComplete()) {
+    throw new Error ("No more frames available, create a new game to play again.");
+  }
+  if((this.currentFrame()[0] !== 10) && (this.currentFrame()[0] + pins > 10)){
+    throw new Error ("Illegal");
+  }
+  this.currentFrame().push(pins);
```

Comment:
refactoring and potentially a private method? 

this.currentFrame().push(pins);
  this.processPreviousFrameBonus(pins);
  this.processPreviousPreviousFrameBonus(pins);